src/features/search/components/SearchHistoryItem/SearchHistoryItem.tsx

```diff
@@ -1,61 +1,27 @@
-import { useNavigation } from '@react-navigation/native'
 import React from 'react'
-import { Keyboard, Text } from 'react-native'
+import { Text } from 'react-native'
 import styled from 'styled-components/native'
-import { v4 as uuidv4 } from 'uuid'
 
-import { SearchGroupNameEnumv2 } from 'api/gen'
-import { UseNavigationType } from 'features/navigation/RootNavigator/types'
-import { getTabNavConfig } from 'features/navigation/TabBar/helpers'
 import { Highlight } from 'features/search/components/Highlight/Highlight'
-import { useSearch } from 'features/search/context/SearchWrapper'
-import { getNativeCategoryFromEnum } from 'features/search/helpers/categoriesHelpers/categoriesHelpers'
-import { Highlighted, HistoryItem, SearchState, SearchView } from 'features/search/types'
-import { useSearchGroupLabel } from 'libs/subcategories'
-import { useSubcategories } from 'libs/subcategories/useSubcategories'
+import { Highlighted, HistoryItem } from 'features/search/types'
 import { Li } from 'ui/components/Li'
 import { ClockFilled } from 'ui/svg/icons/ClockFilled'
 import { Typo, getSpacing } from 'ui/theme'
 
 type Props = {
   item: Highlighted<HistoryItem>
   queryHistory: string
+  onPress: (item: Highlighted<HistoryItem>) => void
 }
 
-export function SearchHistoryItem({ item, queryHistory }: Props) {
-  const { data } = useSubcategories()
-  const { searchState } = useSearch()
-  const { navigate } = useNavigation<UseNavigationType>()
-  const searchGroupLabel = useSearchGroupLabel(
-    item.category ? item.category : SearchGroupNameEnumv2.NONE
+export function SearchHistoryItem({ item, queryHistory, onPress }: Props) {
+  const shouldDisplaySearchGroupOrNativeCategory = Boolean(
+    item.nativeCategoryLabel || item.categoryLabel
   )
-  const nativeCategoryLabel = getNativeCategoryFromEnum(data, item.nativeCategory)?.value
-  const shouldDisplaySearchGroupOrNativeCategory = Boolean(item.nativeCategory || item.category)
-
-  const onPress = () => {
-    Keyboard.dismiss()
-    // When we hit enter, we may have selected a category or a venue on the search landing page
-    // these are the two potentially 'staged' filters that we want to commit to the global search state.
-    // We also want to commit the price filter, as beneficiary users may have access to different offer
-    // price range depending on their available credit.
-    const searchId = uuidv4()
-    const newSearchState: SearchState = {
-      ...searchState,
-      query: item.query,
-      view: SearchView.Results,
-      searchId,
-      isAutocomplete: true,
-      offerGenreTypes: undefined,
-      offerNativeCategories: item.nativeCategory ? [item.nativeCategory] : undefined,
-      offerCategories: item.category ? [item.category] : [],
-    }
-
-    navigate(...getTabNavConfig('Search', newSearchState))
-  }
 
   return (
     <StyledLi>
-      <HistoryItemTouchable onPress={onPress}>
+      <HistoryItemTouchable onPress={() => onPress(item)}>
```

Comment:
J'aime bien avoir un truc du genre 
```ts
const handlePress = useCallback(() => {
  onPress(item)
}, [item, onPress])
```

et donc à l'utilisation
```suggestion
      <HistoryItemTouchable onPress={handlePress}>
```

parce que la fonction anonyme va se régénérer à chaque render.